### PR TITLE
more decode stdout on Python 3

### PR DESCRIPTION
**Pull request by cclauss**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

Python 2 does not make a distinction between bytes and str but Python 3 does. When we make calls to subprocess.Popen() on Python 3 we should stdout.decode("utf8") before performing string manipulations on it.

Like #1890 

Addresses the code review comments at https://github.com/nodejs/node-gyp/pull/1893#pullrequestreview-293495919

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ matrix:
       before_install: nvm install 12
     - name: "Python 3.7 on macOS"
       os: osx
-      #osx_image: xcode11
+      osx_image: xcode11
```
Comment (bnoordhuis) [inline .travis.yml]: This should be a separate commit, I think?

Comment (sam-github) [reply .travis.yml]: Did it perhaps come from an incomplete rebase onto or merge from master? It does seem unrelated to the stdout encoding.

Comment (cclauss) [reply .travis.yml]: It was one of the items mentioned at https://github.com/nodejs/node-gyp/pull/1893#pullrequestreview-293495919

**Comment by rvagg** [comment]
landed in 67dec149
